test/collector.test.ts

```diff
@@ -0,0 +1,351 @@
+            expect(await testToken.balanceOf(remainder.address)).to.equal(2);
+        });
+
+        it('Should withdraw when the remainders address sent is the same as the current one', async () => {
+            const { collector, testToken, remainder } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(3, collector.address);
+
+            await collector.updateRemainderAddress(remainder.address);
+            expect('updateRemainderAddress').to.be.calledOnContract(collector);
+            expect(await testToken.balanceOf(collector.address)).to.equal(0);
+            expect(await testToken.balanceOf(remainder.address)).to.equal(3);
+        });
+
+        it('Should fail when token balance > = remainder', async () => {
+            const { collector, testToken, utilWallet } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(4, collector.address);
+
+            await expect(
+                collector.updateRemainderAddress(utilWallet.address)
+            ).to.be.revertedWith('there is balance to share');
+        });
+
+        it('Should fail when is called by an address that is not the owner', async () => {
+            const { collector, utilWallet } = await loadFixture(
+                prepareAllFixture
+            );
+            const externallyLinkedCollector = collector.connect(
+                utilWallet.address
+            );
+            await expect(
+                externallyLinkedCollector.updateRemainderAddress(
+                    utilWallet.address
+                )
+            ).to.be.revertedWith('only owner can call this');
+        });
+    });
+
+    describe('getBalance', () => {
+        it('Should return 0 if the contract has been just deployed', async () => {
+            const { collector } = await loadFixture(prepareAllFixture);
+            await expect(await collector.getBalance()).to.equal(0);
+        });
+
+        it('Should return 100 after that value has been minted', async () => {
+            const { collector, testToken } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(100, collector.address);
+            await expect(await collector.getBalance()).to.equal(100);
+        });
+    });
+
+    describe('withdraw', () => {
+        it('Should withdraw', async () => {
+            const { collector, testToken, partners } = await loadFixture(
+                prepareAllFixture
+            );
+
+            await testToken.mint(100, collector.address);
+            await collector.withdraw();
+
+            expect(await testToken.balanceOf(partners[0].beneficiary)).to.equal(
+                FIRST_PARTNER_SHARE
+            );
+            expect(await testToken.balanceOf(partners[1].beneficiary)).to.equal(
+                SECOND_PARTNER_SHARE
+            );
```

Comment:
Could you please add expect description wherever you use more than one expect call (at least those of the same types) so that it is clear which one failed if any.

Reply:
I tried to add a custom message for this expect and then make it fail but the massage never appeared. It looks like the should method has this functionality but expect no. So one option is to migrate everything from expect to should. Honestly I think it doesn't worth the effort.

Reply:
Maybe I'm missing something, but checking the [documentation](https://www.chaijs.com/api/bdd/#method_equal) I think the right way to achieve this is
```suggestion
            expect(await testToken.balanceOf(partners[1].beneficiary)).to.equal(
                SECOND_PARTNER_SHARE,
                'Wrong Partner[1] balance'
            );
```

Reply:
Yes, and that's what I did. But it didn't work.

Reply:
I tried the following way and works            
           `expect(
                await testToken.balanceOf(partners[1].beneficiary),
                'Wrong Partner[1] balance'
            ).to.equal(SECOND_PARTNER_SHARE);`

Reply:
Yes, it was working since the beginning. I didn't noticed that the personalized error was concatenated to the test framework message. My bad.